### PR TITLE
[GEP-19] Adapt monitoring configuration

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:
https://github.com/gardener/gardener/pull/9695 changes the contract of how extension can provide monitoring configuration, see https://github.com/gardener/gardener/pull/9695/files#diff-93972179f36fa195b948f0f4979acab75697f99f3c51ab889994d9d8ba9f170b. This PR adapts this extension accordingly. It also cleans up no longer needed configuration (if found).

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9065

**Special notes for your reviewer**:
/cc @ScheererJ 

**Release note**:

```other operator
This extension is now using the new way of providing monitoring configuration (ref [GEP-19](https://github.com/gardener/gardener/blob/master/docs/proposals/19-migrating-observability-stack-to-operators.md)) in case a shoot cluster's Prometheus has been migrated to management via `prometheus-operator`.
```

### CODE REVIEW

**Review by vpnachev — APPROVED** [review]
/lgtm

File: pkg/controller/lifecycle/actuator.go [hunk 1]
```diff
@@ -179,6 +181,8 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extension
 		oidcShootAccessSecret.Secret.Name,
 		generatedSecrets[constants.WebhookTLSSecretName].Name,
 		k8sVersion,
+		// TODO(rfranzke): Delete this after August 2024.
+		a.client.Get(ctx, client.ObjectKey{Name: "prometheus-shoot", Namespace: ex.Namespace}, &appsv1.StatefulSet{}) == nil,
```
Comment (dimityrmirchev) [inline pkg/controller/lifecycle/actuator.go:185]: Shouldn't we return the error if it is different from "not found"? 🤔

Comment (rfranzke) [reply pkg/controller/lifecycle/actuator.go:185]: There is not much value in doing so - we can simply use the "old contract" in that case. This contract is still valid and all this is just to ensure backwards-compatibility. After a few releases, only the "new contract" will be used, and then this code disappears again. Hence, I implemented the "minimal solution" to bridge this time gap.

Comment (dimityrmirchev) [reply pkg/controller/lifecycle/actuator.go:185]: Ok, fair enough.

**Pull request merged**